support/routes/v2/formsg/formsgSiteLaunch.ts

```diff
@@ -204,6 +214,12 @@ export class FormsgSiteLaunchRouter {
       ) || []
 
     res.sendStatus(200) // we have received the form and obtained relevant field
+    this.siteLaunchLogger.setBindings({
+      submissionId,
+      primaryDomain: Array.from(
+        new Set(formResponses.map((x) => x.primaryDomain))
+      ),
+    })
```

Comment:
⚠️ ⚠️ Warning: this is not a correct pattern to apply here 😰 

`FormsgSiteLaunchRouter` is a router class, and so its method `FormsgSiteLaunchRouter.handleSiteLaunchFormRequest()` may be invoked multiple times to handle multiple requests (possibly in parrallel?). We should **not** be storing request-specific data on the logger of the router instance.

The assumption that the logger will be used specifically by the call to `handleSiteLaunchResults()` that follows below does not hold. The method is async and has multiple async tasks therein, and so it is possible that the logger gets modified by a fresher request, and corrupts the logging that is done in the older request 😱 

As tempting as it was to simplify the call sites by having all required information will be logged form the context bindings, we need to ensure correctness first! In this case, I think the call site themselves probably still need to log the information themselves.

OR, you can create a new specific child logger not attached to the instance and pass it into `handleSiteLaunchResults()` below, but that feels a little dirty, since it pollutes the signature of the method 🤔

Reply:
you are absolutely right here, thank you for the catch. 
i have [ccaf5b0](https://github.com/isomerpages/isomercms-backend/pull/1358/commits/ccaf5b0b0c053e0c0986a119f1c32fe7ac0228d8), seems like those only work (usefully) if they are tied within the same function or if the logger is passed down to the various function calls, which does pollute the signature of the method, which does seem too much at this point of time.

Reply:
Thanks Kishore, the change looks good now! 👍 

Not something to spend time on now, but there may be ways to hook into nodejs's context system to store data there. That could allow to store and access pre-configured assets (a child logger in this case) without polluting method signatures.

Looking at the source code for [dd-trace-js](https://github.com/DataDog/dd-trace-js) might be a good source of info on this, since it's basically how tracers "magically" handle contextual scopes (e.g. with `const span = tracer.scope().active()`)

Perhaps a good Learning-Friday activity 😅